docs: add hierarchical-inventory-parent constraint warning

SujalTripathi opened the pull request:
Fixes #16423

Added critical warning documentation for Hierarchical-Inventory-Parent relationship constraints.

## Changes

- Added a prominent warning alert using Meshery's native alert component
- Explained the one-parent-per-component rule with clear examples
- Documented technical consequences: infinite loops, excessive resource creation, system instability
- Provided visual hierarchy examples (correct ✅ vs incorrect ❌)
- Included best practices for using alternative relationships (Edge, Sibling)

## Why This Matters

The video in #16423 demonstrates how multiple parent relationships create cyclical dependencies causing hundreds of duplicate NAT Gateway components to spawn uncontrollably. This warning prevents users from creating such problematic configurations.

## Preview

The warning will appear as a styled yellow/orange alert box immediately after the Hierarchical relationship explanation, making it highly visible to contributors.

Signed-off-by: SujalTripathi <sujaltripathi816@gmail.com>

l5io commented:
🚀 Preview for commit c9becae1e141ee83dacc46c3687a4da15c832280 at: https://69394f2d4efe47342fbe80d9--meshery-docs-preview.netlify.app

leecalcote commented on docs/pages/project/contributing/contributing-model-relationships.md:
```diff
@@ -119,6 +119,8 @@ Relationships can be classified into three main categories:
    1. **Parent**: A parent-child relationship implies the requirement of the parent component before the child component can be created. For example, a `Namespace` in Kubernetes can be a parent of `Pods` within that namespace. The namespace must exist before creating pods within it.
    2. **Inventory**: A hierarchical inventory relationship implies the configuration of a(parent) component is patched with the configuration of other (child) component. For example, Wasm filters can inherit features and functionalities from Envoy filters. This can be used to build on existing functionalities provided by Envoy filters and further extend them using Wasm filters. It enables a modul ...
 
+{% include alert.html type="warning" title="⚠️ Important: Hierarchical-Inventory-Parent Constraint" content="<p><strong>Critical Requirement:</strong> Each component must have <strong>exactly ONE</strong> Hierarchical-Inventory-Parent relationship.</p><p>Adding multiple parent relationships to a single component creates <strong>cyclical dependencies</strong> that result in:</p><ul><li><strong>Inf ...
```
This isn't correct. Children can have more than one parent.

SujalTripathi commented:
@leecalcote Hi Sir! I've addressed your feedback in commit ee886fe.

**Changes made:**
- ✅ Removed incorrect "exactly ONE parent" constraint
- ✅ Updated to focus on avoiding **cyclical dependencies** (A → B → A loops)
- ✅ Changed from "strict tree" to "DAG (Directed Acyclic Graph)" pattern
- ✅ Clarified that components **CAN** have multiple parents, just not circular relationships

The warning now correctly explains that the issue is circular configuration patch loops, not multiple parents.

Could you please review when you have a moment? Thank you!
